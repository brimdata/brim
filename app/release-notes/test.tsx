import {render, screen} from "@testing-library/react"
import {rest} from "msw"
import React from "react"
import server from "test/unit/server"
import ReleaseNotes from "./release-notes"

server.use(
  rest.get(
    "https://api.github.com/repos/brimdata/brim/releases/tags/v0.0.0",
    (req, res, ctx) => {
      return res(ctx.status(200), ctx.json({body: "Testing Release Notes"}))
    }
  )
)

test("fetches the release notes", async () => {
  render(<ReleaseNotes />)
  await screen.findByText("Testing Release Notes")
})
