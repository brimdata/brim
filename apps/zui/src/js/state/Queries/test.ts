import initTestStore from "src/test/unit/helpers/initTestStore"
import Queries from "."
import QueryVersions from "../QueryVersions"
import {makeBuildSelector} from "./selectors"
import {Store} from "../types"

let store: Store
beforeEach(async () => {
  store = await initTestStore()
})

test("build selector", () => {
  store.dispatch(Queries.addItem({name: "My Queriy", id: "1"}))
  store.dispatch(Queries.addItem({name: "My Queriy", id: "2"}))
  store.dispatch(
    QueryVersions.at("1").create({
      ts: new Date().toISOString(),
      version: "0.1",
      value: "count()",
      pins: [{type: "from", value: "mypool"}],
    })
  )
  const build1 = makeBuildSelector()
  const build2 = makeBuildSelector()

  const instance1 = build1(store.getState(), "1")
  const instance2 = build2(store.getState(), "2")
  const instance3 = build1(store.getState(), "1")
  const instance4 = build2(store.getState(), "2")
  expect(instance1).toBe(instance3)
  expect(instance2).toBe(instance4)
})
