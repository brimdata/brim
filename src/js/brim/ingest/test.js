/* @flow */
import ingest from "./"

test("one pcap default", () => {
  let data = ingest.getParams([{type: "pcap", path: "/work/my.pcap"}])

  expect(data).toEqual({
    dataDir: "",
    name: "my.pcap.brim",
    endpoint: "pcap",
    paths: ["/work/my.pcap"]
  })
})

test("one zeek log default", () => {
  let data = ingest.getParams([{type: "log", path: "/work/zeek.log"}])

  expect(data).toEqual({
    name: "zeek.log.brim",
    dataDir: "",
    endpoint: "log",
    paths: ["/work/zeek.log"]
  })
})

test("two zeek logs in same dir default", () => {
  let data = ingest.getParams([
    {type: "log", path: "/work/zeek-1.log"},
    {type: "log", path: "/work/zeek-2.log"}
  ])

  expect(data).toEqual({
    name: "work.brim",
    dataDir: "",
    endpoint: "log",
    paths: ["/work/zeek-1.log", "/work/zeek-2.log"]
  })
})

test("two zeek logs in different dir default", () => {
  let data = ingest.getParams(
    [
      {type: "log", path: "/work/day-1/zeek.log"},
      {type: "log", path: "/work/day-2/zeek.log"}
    ],
    "",
    new Date(0)
  )

  expect(data).toEqual({
    name: "zeek_1969-12-31_16:00:00.brim",
    dataDir: "",
    endpoint: "log",
    paths: ["/work/day-1/zeek.log", "/work/day-2/zeek.log"]
  })
})

test("two pcaps", () => {
  let data = ingest.getParams([
    {type: "pcap", path: "/pcap-1"},
    {type: "pcap", path: "/pcap-2"}
  ])

  expect(data).toEqual({
    error: "Only one pcap can be opened at a time."
  })
})

test("1 pcap and 1 zeek", () => {
  let data = ingest.getParams([
    {type: "pcap", path: "/pcap-1"},
    {type: "log", path: "/zeek-1"}
  ])

  expect(data).toEqual({
    error: "Only one pcap can be opened at a time."
  })
})
