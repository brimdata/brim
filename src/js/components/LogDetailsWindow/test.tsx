/**
 * @jest-environment jsdom
 */

import {lakePath} from "app/router/utils/paths"
import React from "react"
import {viewLogDetail} from "src/js/flows/viewLogDetail"
import {createRecord} from "test/shared/factories/zed-factory"
import fixtures from "test/unit/fixtures"
import {fireEvent, flushPromises, render, screen} from "test/unit/helpers"
import {setupBrim} from "test/unit/helpers/setup-brim"
import LogDetailsWindow from "./index"

const workspace = fixtures("workspace1")
const pool = fixtures("pool1")
const record = createRecord({ts: new Date(0), name: "Alice", age: 99})
const brim = setupBrim({page: "detail", workspace, pool})

beforeEach(async () => {
  brim.dispatch(viewLogDetail(record))
  brim.navTo(lakePath(pool.id, workspace.id))
})

test("right click => new search focuses search window", async () => {
  const {store, main} = brim
  render(<LogDetailsWindow />, {store})

  fireEvent.contextMenu(screen.getByText("Alice"))
  fireEvent.click(screen.getByText(/new search with this value/i))
  await flushPromises()

  const searchWindow = main.windows.getAll().find((w) => w.name === "search")
  expect(searchWindow.ref.focus).toHaveBeenCalled()
})
