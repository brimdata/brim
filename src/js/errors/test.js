/* @flow */
import errors from "./"

test("pcapIngestError", () => {
  let err = "fatal error on backend"

  expect(errors.pcapIngest(err)).toEqual({
    type: "PCAPIngestError",
    message: "Unable to generate full summary logs from PCAP",
    details: ["Detail: fatal error on backend"]
  })
})

test("pcapIngestError (sort limit)", () => {
  let err = "sort limit reached"

  expect(errors.pcapIngest(err)).toEqual({
    type: "PCAPIngestError",
    message: "Unable to generate full summary logs from PCAP",
    details: [
      "Detail: sort limit reached",
      "This PCAP contains too much network traffic to load into Brim."
    ]
  })
})
