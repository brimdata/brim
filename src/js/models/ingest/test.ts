/**
 * @jest-environment jsdom
 */

import {derivePoolName} from "./getParams"

const fakeFile = (path: string, name: string): File => {
  const f = new File([], name)
  f.path = path
  return f
}

test("one pcap default", () => {
  const name = derivePoolName(
    [{type: "pcap", file: fakeFile("/work/my.pcap", "my.pcap")}],
    []
  )
  expect(name).toEqual("my.pcap")
})

test("two zeek logs in same dir default", () => {
  const name = derivePoolName(
    [
      {type: "log", file: fakeFile("/work/zeek-1.log", "zeek-1.log")},
      {type: "log", file: fakeFile("/work/zeek-2.log", "zeek-2.log")},
    ],
    []
  )

  expect(name).toEqual("work")
})
