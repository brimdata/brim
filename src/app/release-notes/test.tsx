/**
 * @jest-environment jsdom
 */

import React from "react"
import {rest} from "msw"
import {SystemTest} from "src/test/system"
import ReleaseNotes from "./release-notes"
import {screen} from "@testing-library/react"

const system = new SystemTest("release-notes")

system.network.use(
  rest.get(
    "https://api.github.com/repos/brimdata/brim/releases/tags/v0.0.0",
    (req, res, ctx) => {
      return res(ctx.status(200), ctx.json({body: "Testing Release Notes"}))
    }
  )
)

test("fetches the release notes", async () => {
  system.render(<ReleaseNotes />)
  await screen.findByText("Testing Release Notes")
})
