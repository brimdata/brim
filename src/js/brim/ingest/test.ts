import ingest from "./"

const fakeFile = (path: string): File => {
  const f = new File([], "fake")
  f.path = path
  return f
}

test("one pcap default", () => {
  const data = ingest.getParams([
    {type: "pcap", file: fakeFile("/work/my.pcap")}
  ])

  expect(data).toEqual({
    dataDir: "",
    name: "my.pcap",
    endpoint: "pcap",
    files: [fakeFile("/work/my.pcap")]
  })
})

test("one zeek log default", () => {
  const data = ingest.getParams([
    {type: "log", file: fakeFile("/work/zeek.log")}
  ])

  expect(data).toEqual({
    name: "zeek.log",
    dataDir: "",
    endpoint: "log",
    files: [fakeFile("/work/zeek.log")]
  })
})

test("two zeek logs in same dir default", () => {
  const data = ingest.getParams([
    {type: "log", file: fakeFile("/work/zeek-1.log")},
    {type: "log", file: fakeFile("/work/zeek-2.log")}
  ])

  expect(data).toEqual({
    name: "work",
    dataDir: "",
    endpoint: "log",
    files: [fakeFile("/work/zeek-1.log"), fakeFile("/work/zeek-2.log")]
  })
})

test("two zeek logs in different dir default", () => {
  const data = ingest.getParams(
    [
      {type: "log", file: fakeFile("/work/day-1/zeek.log")},
      {type: "log", file: fakeFile("/work/day-2/zeek.log")}
    ],
    "",
    [],
    new Date(0)
  )

  expect(data).toEqual({
    name: "zeek_1969-12-31_16:00:00",
    dataDir: "",
    endpoint: "log",
    files: [fakeFile("/work/day-1/zeek.log"), fakeFile("/work/day-2/zeek.log")]
  })
})

test("two pcaps", () => {
  const data = ingest.getParams([
    {type: "pcap", file: fakeFile("/pcap-1")},
    {type: "pcap", file: fakeFile("/pcap-2")}
  ])

  expect(data).toEqual({
    error: "Only one pcap can be opened at a time."
  })
})

test("1 pcap and 1 zeek", () => {
  const data = ingest.getParams([
    {type: "pcap", file: fakeFile("/pcap-1")},
    {type: "log", file: fakeFile("/zeek-1")}
  ])

  expect(data).toEqual({
    error: "Only one pcap can be opened at a time."
  })
})
