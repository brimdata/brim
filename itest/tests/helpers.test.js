/* @flow */

import {basename} from "path"
import {execSync} from "child_process"

import {
  getCopyForCurl,
  getDebugAst,
  logIn,
  newAppInstance,
  openCopyForCurl,
  openDebugQuery,
  resetState,
  setDebugQuery,
  setSpan,
  setSpace,
  startApp,
  startSearch,
  waitForSearch,
  writeSearch
} from "../lib/app.js"
import {handleError, stdTest} from "../lib/jest.js"
import {selectors} from "../../src/js/test/integration"

const extractCurlPost = (command) => JSON.parse(command.split("'")[1])

describe("Debug and search helpers", () => {
  let app
  let testIdx = 0
  beforeEach(() => {
    app = newAppInstance(basename(__filename), ++testIdx)
    return startApp(app)
  })

  afterEach(async () => {
    if (app && app.isRunning()) {
      await resetState(app)
      return await app.stop()
    }
  })

  stdTest("fresh state shows default Debug Query AST", (done) => {
    logIn(app)
      .then(() => openDebugQuery(app))
      .then(() => getDebugAst(app))
      .then((searchResults) => {
        expect(searchResults).toMatchSnapshot()
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })

  stdTest(
    "write into main search shows Debug Query AST (update after fixing PROD-950 again and remove this clause)",
    (done) => {
      logIn(app)
        .then(() =>
          writeSearch(
            app,
            "_path=x509 | every 1d count() by certificate.version | sort ts, certificate.version"
          )
        )
        .then(() => openDebugQuery(app))
        .then(() => getDebugAst(app))
        .then((searchResults) => {
          expect(searchResults).toMatchSnapshot()
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    }
  )

  stdTest("write new search into Debug Query shows AST", (done) => {
    logIn(app)
      .then(() => openDebugQuery(app))
      .then(() =>
        setDebugQuery(
          app,
          "_path=x509 | every 1d count() by certificate.version | sort ts, certificate.version"
        )
      )
      .then(() => getDebugAst(app))
      .then((searchResults) => {
        expect(searchResults).toMatchSnapshot()
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })

  stdTest(
    "write invalid search into Debug Query shows reasonable error",
    (done) => {
      logIn(app)
        .then(() => openDebugQuery(app))
        .then(() => setDebugQuery(app, "* | nope()"))
        .then(() => app.client.getText(selectors.debugSearch.astError))
        .then((searchResults) => {
          expect(searchResults).toMatchSnapshot()
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    }
  )

  stdTest("fresh state shows default, well-formed Copy for curl", (done) => {
    logIn(app)
      .then(() => openCopyForCurl(app))
      .then(() => getCopyForCurl(app))
      .then((curlCommand) => {
        // curl must be first, and the -d ' must be on the top line for the paste
        // to work cleanly
        expect(curlCommand).toMatch(/^\s*curl[ \t]+[^\n]*-d[ \t]*'/)
        // -X POST must be concurrent non-whitespace tokens that MAY be at the
        // end
        expect(curlCommand).toMatch(/\s+-X\s+POST(?:\s+|$)/)
        // URL MAY be at the end
        expect(curlCommand).toMatch(/\s+http:\/\/[^/]+\/search(?:\s+|$)/)
        expect(extractCurlPost(curlCommand)).toBeTruthy()
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })

  stdTest("exec a curl returns process success", (done) => {
    logIn(app)
      .then(() => writeSearch(app, "* | count()"))
      .then(() => startSearch(app))
      .then(() => waitForSearch(app))
      .then(() => openCopyForCurl(app))
      .then(() => getCopyForCurl(app))
      .then((curlCommand) => {
        try {
          execSync(curlCommand, {stdio: "pipe"})
        } catch (err) {
          done(err)
        }
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })

  stdTest(
    "write into main search shows proper AST in Copy for curl command (update after fixing PROD-950 again and remove this clause)",
    (done) => {
      logIn(app)
        .then(() =>
          writeSearch(
            app,
            "_path=x509 | every 1d count() by certificate.version | sort ts, certificate.version"
          )
        )
        .then(() => openCopyForCurl(app))
        .then(() => getCopyForCurl(app))
        .then((curlCommand) => {
          expect(extractCurlPost(curlCommand)).toMatchSnapshot()
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    }
  )

  stdTest("space change reflected in Copy for curl command", (done) => {
    logIn(app)
      .then(() => setSpace(app, "hq_integration"))
      .then(() => openCopyForCurl(app))
      .then(() => getCopyForCurl(app))
      .then((curlCommand) => {
        expect(extractCurlPost(curlCommand).space).toBe("hq_integration")
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })

  stdTest("span change reflected in Copy for curl command", (done) => {
    logIn(app)
      .then(() => setSpan(app, "Whole Space"))
      .then(() => openCopyForCurl(app))
      .then(() => getCopyForCurl(app))
      .then((curlCommand) => {
        expect(extractCurlPost(curlCommand).span.dur.sec).toBe(3352171)
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })
})
