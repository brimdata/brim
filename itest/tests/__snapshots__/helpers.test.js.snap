// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debug and search helpers fresh state shows default Debug Query AST 1`] = `"{\\"proc\\":{\\"op\\":\\"SourceProc\\",\\"filter\\":{\\"op\\":\\"BooleanLiteral\\",\\"value\\":true}}}"`;

exports[`Debug and search helpers write into main search shows Debug Query AST (update snapshot after fixing PROD-950) 1`] = `"{\\"proc\\":{\\"op\\":\\"SourceProc\\",\\"filter\\":{\\"op\\":\\"BooleanLiteral\\",\\"value\\":true}}}"`;

exports[`Debug and search helpers write invalid search into Debug Query shows reasonable error 1`] = `"SyntaxError: Expected \\"=\\", [A-Za-z0-9_.], or whitespace but \\"(\\" found."`;

exports[`Debug and search helpers write new search into Debug Query shows AST 1`] = `"{\\"proc\\":{\\"op\\":\\"SequentialProc\\",\\"procs\\":[{\\"op\\":\\"SourceProc\\",\\"filter\\":{\\"op\\":\\"CompareField\\",\\"comparator\\":\\"eql\\",\\"field\\":{\\"op\\":\\"FieldRead\\",\\"field\\":\\"_path\\"},\\"value\\":{\\"type\\":\\"string\\",\\"value\\":\\"x509\\"}}},{\\"op\\":\\"GroupByProc\\",\\"keys\\":[\\"certificate.version\\"],\\"reducers\\":[{\\"op\\":\\"Count\\",\\"var\\":\\"count\\"}],\\"duration\\":{\\"type\\":\\"Duration\\",\\"seconds\\":86400}},{\\"op\\":\\"SortProc\\",\\"fields\\":[\\"ts\\",\\"certification.version\\"],\\"sortdir\\":1}]}}"`;
